File: CodeGeneratorMVC/CommonObjects/Hasher.cs
using System.Text;
using Microsoft.VisualBasic;

namespace Tools {
    public class Hasher {
        /// <summary>
		/// 		''' create SHA1 Hash from given String
		/// 		''' </summary>
		/// 		''' <param name="hashString">String to Hash</param>
		/// 		''' <returns>40 Character Hex Hash for given string.</returns>
		/// 		''' <remarks></remarks>
        public static string CreateSHA1Hash(string hashString) {
            byte[] data = System.Text.Encoding.ASCII.GetBytes(hashString);
            System.Security.Cryptography.SHA1 hasher;
            hasher = System.Security.Cryptography.SHA1.Create();
            byte[] hash = hasher.ComputeHash(data);
            StringBuilder stB = new StringBuilder();
            foreach (byte b in hash)
                stB.Append(Conversion.Hex(b).PadLeft(2, '0'));
            return stB.ToString();
        }
    }
}

File: CodeGeneratorMVC/oldClasses/CommonObjects/Hasher.vb
Imports Microsoft.VisualBasic
Imports System.Text

Namespace Tools
	Public Class Hasher
		''' <summary>
		''' create SHA1 Hash from given String
		''' </summary>
		''' <param name="hashString">String to Hash</param>
		''' <returns>40 Character Hex Hash for given string.</returns>
		''' <remarks></remarks>
		Public Shared Function CreateSHA1Hash(ByVal hashString As String) As String
			Dim data() As Byte = System.Text.Encoding.ASCII.GetBytes(hashString)
			Dim hasher As System.Security.Cryptography.SHA1
			hasher = System.Security.Cryptography.SHA1.Create()
			Dim hash() As Byte = hasher.ComputeHash(data)
			Dim stB As New StringBuilder
			For Each b As Byte In hash
				stB.Append(Hex(b).PadLeft(2, "0"c))
			Next b
			Return stB.ToString
		End Function
	End Class
End Namespace
